src/autolabel/few_shot/label_selector.py

```diff
@@ -13,45 +17,56 @@ class LabelSelector:
     labels: List[str]
     """A list of the possible labels to choose from."""
 
-    k: int = 10
-    """Number of labels to select"""
-
-    embedding_func: Callable = None
-    """Function used to generate embeddings of labels/input"""
+    label_descriptions: Optional[Dict[str, str]]
+    """A dictionary of label descriptions. If provided, the selector will
+    use these descriptions to find the most similar labels to the input."""
 
     labels_embeddings: Dict = {}
     """Dict used to store embeddings of each label"""
 
+    cache: bool = True
+    """Whether to cache the embeddings of labels"""
+
     def __init__(
-        self, labels: List[str], embedding_func: Callable, k: int = 10
+        self,
+        config: Union[AutolabelConfig, str, dict],
+        embedding_func: Callable,
+        cache: bool = True,
     ) -> None:
-        self.labels = labels
-        self.k = min(k, len(labels))
-        self.embedding_func = embedding_func
-        for l in self.labels:
-            self.labels_embeddings[l] = self.embedding_func.embed_query(l)
+        self.config = config
+        self.labels = self.config.labels_list()
+        self.label_descriptions = self.config.label_descriptions()
+        self.k = min(self.config.label_selection_count(), len(self.labels))
+        self.threshold = self.config.label_selection_threshold()
+        self.cache = cache
+        self.vectorStore = VectorStoreWrapper(
+            embedding_function=embedding_func, cache=self.cache
+        )
+
+        # Get the embeddings of the labels
+        if self.label_descriptions is not None:
+            (labels, descriptions) = zip(*self.label_descriptions.items())
+            embeddings = self.vectorStore._get_embeddings(descriptions)
+            for i, label in enumerate(labels):
+                self.labels_embeddings[label] = embeddings[i]
+        else:
+            embeddings = self.vectorStore._get_embeddings(self.labels)
+            for i, label in enumerate(labels):
+                self.labels_embeddings[label] = embeddings[i]
 
     def select_labels(self, input: str) -> List[str]:
         """Select which labels to use based on the similarity to input"""
-        input_embedding = self.embedding_func.embed_query(input)
+        input_embedding = self.vectorStore._get_embeddings([input])
 
         scores = []
         for label, embedding in self.labels_embeddings.items():
             similarity = cos_sim(embedding, input_embedding)
             # insert into scores, while maintaining sorted order
             bisect.insort(scores, (similarity, label))
```

Comment:
lets do this in a vectorized way? not sure how much of a performance hit doing this is